### PR TITLE
Fix order of dwrf encryption group stats

### PR DESCRIPTION
**Pull request by rschlussel**
Test plan - Added unit test for this case and added stats verification for all decryption tests

```
== RELEASE NOTES ==
Hive Changes
* Fix a bug where non-Presto readers could not read encrypted DWRF files written by Presto if the encryption group listed columns out of order.
```

### CODE REVIEW

**Review by mayankgarg1990 — APPROVED** [review]

File: presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java [hunk 1]
```diff
@@ -610,19 +610,22 @@ public void close()
...
             for (int i = 0; i < writerEncryptionGroups.size(); i++) {
                 WriterEncryptionGroup group = writerEncryptionGroups.get(i);
+                int groupId = i;
                 encryptionGroupBuilder.add(
                         new EncryptionGroup(
                                 group.getNodes(),
                                 Optional.empty(), // reader will just use key metadata from the stripe
-                                encryptedStats.get(i)));
+                                group.getNodes().stream()
+                                        .map(nodeId -> encryptedStats.get(groupId).get(nodeId))
```
Comment (sdruzkin) [inline presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java:627]: You can optimize this piece a bit by extracting and moving `encryptedStats.get(groupId)` outside to line 621.

File: presto-orc/src/test/java/com/facebook/presto/orc/TestDecryption.java [hunk 2]
```diff
@@ -311,6 +312,39 @@ public void testSingleEncryptionGroupRowType()
                 outputColumns);
     }
 
+    @Test
+    public void testEncryptionGroupWithReversedOrderNodes()
```
Comment (sdruzkin) [inline presto-orc/src/test/java/com/facebook/presto/orc/TestDecryption.java:316]: Can we also add the same test but with the direct order? I briefly looked through other tests cases, they all seems to have the same type for all columns.

Comment (rschlussel) [reply presto-orc/src/test/java/com/facebook/presto/orc/TestDecryption.java:349]: there are a couple tests that have multiple types (e.g. https://github.com/prestodb/presto/blob/master/presto-orc/src/test/java/com/facebook/presto/orc/TestDecryption.java#L315), but it's good to have a simple explicit test, so added that case.

**Review by sdruzkin — APPROVED** [review]

**Review by mbasmanova — APPROVED** [review]

**Review by bhhari — APPROVED** [review]
LG

**Pull request merged**